### PR TITLE
Write Saved WiFi networks and their passwords in a text file in C

### PR DESCRIPTION
**Pull request by Nibba2018**
Parent issue: #220 

### Type of change

**Select all that applies**:
* [X] New feature (non-breaking change which adds functionality)

### Checklist:

* [X] My code follows the style guidelines(Clean Code) of this project.
* [X] I have performed a self-review of my own code.
* [X] I have used consistent code conventions and clear function/method/variable names.
* [X] I have commented on my code, particularly in hard-to-understand areas.
* [X] I have created a helpful and easy to understand README.md which contains description, requirements and use case.
* [X] I have added relevent screenshots/images if necessary in a seperate folder.
* [X] I have added my work in respective language folder.
* [X] My changes generate no new warnings.

### CODE REVIEW

**Comment by SANKET7738** [comment]
please squash the commits @Nibba2018

**Comment by Nibba2018** [comment]
@SANKET7738 Done.

File: C/Wifi-Passwords/gen-wifi.c [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        char *ssid = strtok(buffer, ":");
+        ssid = strtok(NULL, ": ");
+
+        // Trim trailing whitespaces.
+        char *end;
+        end = ssid + strlen(ssid) - 1;
+        while(end > ssid && isspace((unsigned char)*end)) end--;
+        end[1] = '\0';
+
+
```
Comment (yashaswibiyahut) [inline C/Wifi-Passwords/gen-wifi.c]: kindly remove all the extra space

File: C/Wifi-Passwords/gen-wifi.c [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        fprintf(fptr, "%s", formattedLog);
+    }
+
+
+    // Close all file pointers and free up memory space.
+    pclose(shell_op);
+    fclose(fptr);
+
+    return 0;
+}
```
Comment (yashaswibiyahut) [inline C/Wifi-Passwords/gen-wifi.c:151]: add space and time complexity of your program.

**Review by yashaswibiyahut — CHANGES_REQUESTED** [review]

File: C/Wifi-Passwords/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+## Executing the Program:
+* Compile the program: `gcc gen-wifi.c`.
+* On windows, do `a.exe`.
+* On Linux, do `./a.out`.
+
+**Contents of `wifi_password.txt`:**
+```txt
+<SSID 1>:<password>
+<SSID 2>:<password>
+.
```
Comment (yashaswibiyahut) [inline C/Wifi-Passwords/README.md:16]: Add a sample use case /example of your code in readme file.

**Review by yashaswibiyahut — CHANGES_REQUESTED** [review]

File: C/Wifi-Passwords/README.md (hunk 3 above)
Comment (Nibba2018) [reply C/Wifi-Passwords/README.md:16]: @yashaswibiyahut done.

**Review by yashaswibiyahut — APPROVED** [review]

**Review by SANKET7738 — CHANGES_REQUESTED** [review]
I can still see 3 commits. Please squash them into a single one. @Nibba2018

**Comment by Tejas1510** [comment]
@SANKET7738 @yashaswibiyahut is the issue only restricted to c language or can I work in python?
If possible to work in python please assign me.

**Comment by yashaswibiyahut** [comment]
@Tejas1510  kindly check the parent issue.
For python this issue is already merged.

**Comment by Nibba2018** [comment]
@yashaswibiyahut @SANKET7738 Is there any other changes that you would like to have?

**Pull request merged**